src/MSBuild/XMake.cs

```diff
@@ -1427,17 +1532,29 @@ private static void ResetGatheringSwitchesState()
                             {
                                 if (graphBuildOptions != null)
                                 {
-                                    (result, exception) = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    graphResult = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    ProjectGraphEntryPoint entryPoint = graphBuildRequest.ProjectGraphEntryPoints.First();
```

Comment:
There should only ever be exactly one of these in this codepath, right?

```suggestion
                                    ProjectGraphEntryPoint entryPoint = graphBuildRequest.ProjectGraphEntryPoints.Single();
```